pkg/runtime/runtime.go

```diff
@@ -54,17 +57,24 @@ type TemplateSpec struct {
 	// ObjApply is ApplyConfiguration for the TrainingRuntimes Template field.
 	ObjApply any
 	// PodSets is a set of Pod extracted from ObjApply.
+	// This is abstract concept to represent multiple PodSpec as a unit.
 	PodSets []PodSet
 }
 
 type PodSet struct {
+	// PodSet name is the name to identify PodSpec.
+	// This typically has the name stored in each PodSpec.
 	Name string
 	// If Name is trainer-node, CountForNonTrainer is null.
 	// For Trainer, PodSet Count should be stored in Info.RuntimePolicy.MLPolicy.NumNodes.
 	CountForNonTrainer *int32
```

Comment:
Since we already removed the Trainer struct, do you want to refactor this count parameter in the next PR ?
I just feel like instead of updating the `info.RuntimePolicy.MLPolicy.NumNodes`, we can directly update count value in PodSet for the appropriate name (e.g. Launcher, Node): https://github.com/kubeflow/trainer/blob/e627f78689baff0171f542b816d8943fb5df2bd9/pkg/runtime/framework/plugins/plainml/plainml.go#L56.

Which means PodSet.Count always dictates values for Parallelism and Completions.

Reply:
I feel like it should simplifies a lot of logic we have.
For example: https://github.com/kubeflow/trainer/blob/e627f78689baff0171f542b816d8943fb5df2bd9/pkg/runtime/framework/plugins/coscheduling/coscheduling.go#L128-L131

Reply:
Yes, that's right. I will change MLPolocy data structure only for internal in the next PR.
I decoupled the changes from this PR since the internal MLPololicy data structure change has a lot of affection.

Reply:
>Yes, that's right. I will change MLPolocy data structure only for internal in the next PR.

Do you mean after we refactor MLPolicy for the Info object, we can change the Count value in PodSet ?

Reply:
I will drop MLPolicy.NumNodes only from internal data structure, and rename PodSet.CoutNotForTrainer with PodSet.Count.

Reply:
Yeah, that make sense!